pkg/controller/servicebindingrequest/binder.go

```diff
@@ -493,40 +529,44 @@ func (b *binder) update(objs *unstructured.UnstructuredList) ([]*unstructured.Un
 	updatedObjs := []*unstructured.Unstructured{}
 
 	for _, obj := range objs.Items {
-		// modify the copy of the original object and use the original one later for comparison
-		updatedObj := obj.DeepCopy()
+		// store a copy of the original object to later be used in a comparison
+		originalObj := obj.DeepCopy()
 		name := obj.GetName()
 		log := b.logger.WithValues("Obj.Name", name, "Obj.Kind", obj.GetKind())
 		log.Debug("Inspecting object...")
 
-		sbrNamespacedName := types.NamespacedName{Namespace: b.sbr.GetNamespace(), Name: b.sbr.GetName()}
-		updatedObj = setSBRAnnotations(sbrNamespacedName, updatedObj)
+		var updatedObj *unstructured.Unstructured
+		var err error
+		if b.sbr.Spec.ApplicationSelector.BindingPath.CustomSecretPath != nil {
+			_, err = b.updateSecretField(&obj)
+			if err != nil {
+				return nil, err
+			}
+		}
 
-		updatedObj, err := b.updateSpecContainers(updatedObj)
+		updatedObj, err = b.updateSpecContainers(&obj)
```

Comment:
To make the API coherent, `obj` should not be mutated in `updateSpecContainers`; it is a matter of create a deep copy of the input `obj` and return the modified copy.

The subtlety of this is that the input list is modified in an at-distance action, inside a function that is apparently harmless since its signature indicates a copy is being returned by the statement being executed.